### ISSUE TITLE
Handle 404 when updating a device registration

### ISSUE THREAD

**Issue by tcard** [ably/docs#326]
When the push activation state machine updates the registration and it encounters a 404, that means the device got deregistered somehow, probably through the admin API. We still should call the activation callback with the error, but we also should reset all local device data and move to `NotActivated`.

Part of https://github.com/ably/wiki/issues/161.

┆Issue is synchronized with this [Jira Task](https://ably.atlassian.net/browse/DOC-179) by [Unito](https://www.unito.io/learn-more)

**Comment by tbedford** [comment]
It's not clear if this is a doc issue or not. If a docs issue, please provide a link to the relevant docs. Thanks.

**Comment by owenpearson** [comment]
@kavalerov yeah it's definitely a feature spec improvement suggestion. All the context is there if you follow the link to the ideas issue (and the subsequent link to [this conversation](https://github.com/ably/realtime/pull/993#issuecomment-297006955) which seems to be where the idea originated from.

**Comment by kavalerov** [comment]
@tbedford at the moment feature spec changes also live in the docs repo, so I think this means we need to reopen this issue for now

**Comment by kavalerov** [comment]
Big question is how we manage this moving forward, we probably can separate the spec from the actual documentation, because they serve two very different audiences. Should we maybe move the spec into a separate repo? 

Or, we just keep things as they are and just use correct labels

**Comment by tcard** [comment]
> because they serve two very different audiences

Just my two cents, but: I believe the spec is valuable documentation for end users too, not just for SDK developers. It states exactly what the library does in a terse and exhaustive way that complements the more verbose, and not as deep, main documentation.

**Comment by kavalerov** [comment]
This is a good point @tcard. It would be good to see if customers are using the spec for this - right now we definitely don't make it easy to find, actually wise versa, it is pretty hidden in the docs. 

I think the kind of exhaustive description of how the library works can potentially be part of the docs, but I am a bit worried about content serving two different audiences at the same time, as it will impact the decisions we make and tradeoffs we need to agree with (to be clear - I am not saying it is not valuable for both, I am saying that it is a problem that both can potentially use it).
